From eca67e562870f619efaaf960e27b7d232fc6acea Mon Sep 17 00:00:00 2001
From: Chun-wei Fan <fanchunwei@src.gnome.org>
Date: Tue, 21 Mar 2017 16:36:21 +0800
Subject: [PATCH] gettext-tools/src: Fix linking

xgettext.c needs to be built as C++, so we need to ensure that the
symbols from read-po.h, write-properties.h and write-stringtable.h are
decorated with extern "C" so that they will not be mangled with C++
stuff, enabling the code to link.
---
 gettext-tools/src/read-po.h           | 6 ++++++
 gettext-tools/src/write-properties.h  | 6 ++++++
 gettext-tools/src/write-stringtable.h | 6 ++++++
 3 files changed, 18 insertions(+)

diff --git a/gettext-tools/src/read-po.h b/gettext-tools/src/read-po.h
index 064e27a..c72b432 100644
--- a/gettext-tools/src/read-po.h
+++ b/gettext-tools/src/read-po.h
@@ -21,6 +21,12 @@
 #include "read-catalog-abstract.h"
 
 /* Describes a .po / .pot file parser.  */
+#ifdef __cplusplus
+extern "C" {
+#endif
 extern DLL_VARIABLE const struct catalog_input_format input_format_po;
+#ifdef __cplusplus
+}
+#endif
 
 #endif /* _READ_PO_H */
diff --git a/gettext-tools/src/write-properties.h b/gettext-tools/src/write-properties.h
index cf6f86c..c621b27 100644
--- a/gettext-tools/src/write-properties.h
+++ b/gettext-tools/src/write-properties.h
@@ -21,6 +21,12 @@
 #include "write-catalog.h"
 
 /* Describes a PO file in Java .properties syntax.  */
+#ifdef __cplusplus
+extern "C" {
+#endif
 extern DLL_VARIABLE const struct catalog_output_format output_format_properties;
+#ifdef __cplusplus
+}
+#endif
 
 #endif /* _WRITE_PROPERTIES_H */
diff --git a/gettext-tools/src/write-stringtable.h b/gettext-tools/src/write-stringtable.h
index bf3de6c..179f8ae 100644
--- a/gettext-tools/src/write-stringtable.h
+++ b/gettext-tools/src/write-stringtable.h
@@ -21,6 +21,12 @@
 #include "write-catalog.h"
 
 /* Describes a PO file in .strings syntax.  */
+#ifdef __cplusplus
+extern "C" {
+#endif
 extern DLL_VARIABLE const struct catalog_output_format output_format_stringtable;
+#ifdef __cplusplus
+}
+#endif
 
 #endif /* _WRITE_STRINGTABLE_H */
-- 
2.10.2.windows.1

